Firestore rules documentation

rramirez-dev opened the pull request:
# What it Does
* Closes #146
* Added descriptions for each firestore collection
* Added a table that described the fields of each collection -Added the firebase rules for each collection

# How I Tested
* Add a list of steps to show the functionality of your feature
For example:
* Run the application
* Tap the (+) button
* See this screen
etc...

# Notes
* Anything else that should be noted about how you implemented this feature?

# Screenshot
* Add a screenshot of your new feature! OR show a screen recording of it in action. (On the simulator press Cmd + R to record, and the stop button when done). **This video should be no longer than 30 seconds.**

rramirez-dev committed:
Firestore rules documentation

-Added descriptions for each firestore collection
-Added a table that described the fields of each collection
-Added the firebase rules for each collection

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/Documentation.md:
```diff
@@ -16,3 +16,76 @@ Icons have the following colors:
+
+| Fields    | Type    |
+| --------- | ------- |
+| actionTex | Text    |
```
```suggestion
| actionText | Text    |
```

This was a typo on my end, my bad, please fix this

mikaelacaron replied:
For "Type" are you meaning the Firestore type, or the Swift types? I am guessing Firestore, but can you write that explicitly somewhere?

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/Documentation.md:
```diff
@@ -16,3 +16,76 @@ Icons have the following colors:
+
+## Firestore Collections
+
+**_alerts_**  
```
Instead of this just being italic, you could make it a heading 3 or 4

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/Documentation.md:
```diff
@@ -16,3 +16,76 @@ Icons have the following colors:
+## Firestore Collections
+
+**_alerts_**  
+The alerts collection contains system level alerts that will be visable to all users
```
```
The alerts collection contains system level alerts that will be visable to all users
```

More so than this, can you also note that they will only be shown to the user once

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/Documentation.md:
```diff
@@ -16,3 +16,76 @@ Icons have the following colors:
+**_alerts_**  
+The alerts collection contains system level alerts that will be visable to all users
+
+| Fields    | Type    |
```
Can you add a description too, for that that property does? thanks!

rramirez-dev replied:
What's the difference between actionText and message. What is actionURL for?

mikaelacaron replied:
The `actionText` is the text for the button, the `actionURL` is url that the user will go to when they tap the button

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/Documentation.md:
```diff
@@ -16,3 +16,76 @@ Icons have the following colors:
+        allow read;
+        }
+
+        match /maintainance+events/{allPaths=**} {
+        allow read, write: if request.auth != null;
+        }
```
This allows anyone to read anyone's data, right?
you should only be allowed to read your own data

mikaelacaron replied:
this hasn't changed, is this the correct syntax? Can you link the Firestore documentation for this? @rramirez-dev

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/Documentation.md:
```diff
@@ -16,3 +16,76 @@ Icons have the following colors:
+        allow read, write: if request.auth != null;
+        }
+
+        match /vehicles/{allPaths=**} {
+        allow read, write: if request.auth != null;
+        }
```
Same here, about only being able to read your own data, and on one elses

mikaelacaron reviewed (changes requested):
Great job! I like the description of all the fields

Also note how I filled out the PR template, you should add the info under the proper heading and link the issue you're working on, as shown in the PR template

Can you make this whole thing into a new document called `Firestore Details` or something like that?
And then link to it in this main documentation page?

If you directly fix an issue and don't have any questions, go ahead and click resolve on my comment. If you have questions, comment directly on that thread, once everything's resolved, please click the re-review button 
<img width="345" alt="Re-review button" src="https://github.com/mikaelacaron/Basic-Car-Maintenance/assets/22946902/eba4015d-954f-4d51-bfba-2b474163c040">

mikaelacaron reviewed (changes requested):
Great job! I like the description of all the fields

Can you make this whole thing into a new document called `Firestore Details` or something like that?
And then link to it in this main documentation page?

If you directly fix an issue and don't have any questions, go ahead and click resolve on my comment. If you have questions, comment directly on that thread, once everything's resolved, please click the re-review button 
<img width="345" alt="Re-review button" src="https://github.com/mikaelacaron/Basic-Car-Maintenance/assets/22946902/eba4015d-954f-4d51-bfba-2b474163c040">

rramirez-dev commented:
How would I create a new document in GitHub and link it in the documentation md if I don't know that path?

mikaelacaron commented:
@rramirez-dev You don't have access to Xcode right? Because of versioning with your Mac?

rramirez-dev commented:
I bought a new Mac so I no longer have any Xcode issues. Do I have to make the changes in my branch and do a new PR?

mikaelacaron commented:
No you need to open Xcode, make a new file in the right place, and build the documentation, you do this all on your own branch, make changes, commit them, and then push to your branch on you fork.

And doing that automatically updates the PR

rramirez-dev committed:
Create FirestoreDetails.md

As per PR review I have created a new file called FirestoreDetilas.md to provide information details about the Firestore database.

rramirez-dev committed:
Update Documentation.md

As per PR review I have move the Firestore details to a separate file called FirestoreDetails.md and add a link to the file in the Documentation.md file

rramirez-dev committed:
Update Documentation.md

Fixed a type in the relative link to the FirestoreDetails.md file

rramirez-dev committed:
Merge branch 'mikaelacaron:dev' into 146-firebase-security-rules

rramirez-dev commented:
I have made all the changes on my forked branch and committed them. Do I have to a push on terminal from my branch or is there an interface on GitHub to push my forked branch?

rramirez-dev commented:
I think that my changes were automatically pushed from my branch since I did all my changes from the GitHub interface.

mikaelacaron commented:
I'll take a look later
> Do I have to a push on terminal from my branch or is there an interface on GitHub to push my forked branch

You make the change locally, push your changes to your fork, and that's it. the PR updates automatically

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/Documentation.md:
```diff
@@ -16,3 +16,5 @@ Icons have the following colors:
 * Black #000000
 * Orange #F78616
+
+[Firestore Details](FirestoreDetails.md)
```
This didn't link properly when I click on it, and I can't find the actual documentation page when I build it?

rramirez-dev replied:
When I click the link on my forked branch it works.

mikaelacaron replied:
Yours doesn't show up on the left in articles, and I would expect it to
<img width="311" alt="Screenshot 2023-10-23 at 10 50 19 PM" src="https://github.com/mikaelacaron/Basic-Car-Maintenance/assets/22946902/b77748b4-9082-4afc-8120-e46f6c1db6a7">

Like this one does for the App store stuff
<img width="393" alt="Screenshot 2023-10-23 at 10 51 46 PM" src="https://github.com/mikaelacaron/Basic-Car-Maintenance/assets/22946902/548f44b7-6c6e-40b0-8bfd-bcac4f4fd6a6">

I found the reason for this is because you did not properly format the document, you used a header 2 rather than a header 1

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/FirestoreDetails.md:
```diff
@@ -0,0 +1,79 @@
+
+### _alerts_
+
+The alerts collection contains system level alerts that will be visable to all users. The alerts
```
```suggestion
The alerts collection contains system level alerts that will be visible to all users. The alerts
```

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/FirestoreDetails.md:
```diff
@@ -0,0 +1,79 @@
+| Fields     | Firestore Type | Description                                                                            |
+| ---------- | -------------- | -------------------------------------------------------------------------------------- |
+| actionText | Text           | Indicates the text that will appear on the action button                               |
+| actionURL  | Text           | Indicates the url that the user will be redirected to when the acton button is pressed |
```
```suggestion
| actionURL  | Text           | Indicates the url that the user will be redirected to when the acton button is tapped |
```

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/FirestoreDetails.md:
```diff
@@ -0,0 +1,79 @@
+| actionURL  | Text           | Indicates the url that the user will be redirected to when the acton button is pressed |
+| emojiIcon  | Text           | The emoji that will appear on the alert message                                        |
+| id         | Text           | Identifies a specific alert                                                            |
+| isOn       | Boolean        | Indicates if the alert is active                                                       |
```
```suggestion
| isOn       | Boolean        | Indicates if the alert is active. Only one alert will be active at a time.                                                       |
```

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/FirestoreDetails.md:
```diff
@@ -0,0 +1,79 @@
+| message    | Text           | The text content of the alert                                                          |
+| title      | Text           | The name given to the alert                                                            |
+
+### _maintainance_events_
```
No need to make this italic, just make it a regular heading

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/FirestoreDetails.md:
```diff
@@ -0,0 +1,79 @@
+| notes     | Text           | Extra details that the user want to provide about the maintainance event                  |
+| vehicleID | Text           | Identifies the vehicle assocated with the maintainance event                              |
+
+### _vehicles_
```
Same for this one about being italic

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/FirestoreDetails.md:
```diff
@@ -0,0 +1,79 @@
+
+| name | Text | Customized name given to vehicle entry
+| make | Text | Indicates the brand of the vehicle
+| model | Text | Indicates unique name associated with a specific brand
+| year | Text | Indicates the year the vehicle was released by the manufacturer
+| color | Text | Indicates the manifacturer color associated with the vehicle
```
Instead of using the word "indicates" in front of a lot of properties, you can leave the word off (can you change all the properties this affects

mikaelacaron reviewed (changes requested):
Just a few extra comments and this will be ready!

Once you go through everything please click the re-review button

rramirez-dev committed:
Merge branch 'mikaelacaron:dev' into 146-firebase-security-rules

rramirez-dev committed:
PR-187-Review fixes

    -Fixed spelling errors
    -Fixed type descriptions
    -Removed italic type from collection names
    -Fixed table formatting

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/Documentation.md:
```diff
@@ -16,3 +16,5 @@ Icons have the following colors:
 * Black #000000
 * Orange #F78616
+
+[Firestore Details](FirestoreDetails.md)
```
```suggestion
<doc:FirestoreDetails>
```

This still didn't work for me

This was the proper syntax, found from [this example](https://developer.apple.com/documentation/xcode/slothcreator_building_docc_documentation_in_xcode)

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/FirestoreDetails.md:
```diff
@@ -0,0 +1,78 @@
+| id        | DocumentID     | Unique maintainance event identifier that will auto-populate your objects id property |
+| userID    | Text           | Identifies the user associated with the maintainance event                                |
+| title     | Text           | Customized name given to the maintainance event                                       |
+| date      | Date           | Date that the maintainance event was created                                          |
```
```suggestion
| date      | Date           | Date that the maintenance event was recorded                                          |
```

Going to word it this way because technically it's not the date it's created, but whatever date when someone decides to enter it

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/FirestoreDetails.md:
```diff
@@ -0,0 +1,78 @@
+| userID    | Text           | Identifies the user associated with the maintainance event                                |
+| title     | Text           | Customized name given to the maintainance event                                       |
+| date      | Date           | Date that the maintainance event was created                                          |
+| notes     | Text           | Extra details that the user want to provide about the maintainance event                  |
```
```suggestion
| notes     | Text           | Extra details about the maintenance event                  |
```

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/FirestoreDetails.md:
```diff
@@ -0,0 +1,78 @@
+| title     | Text           | Customized name given to the maintainance event                                       |
+| date      | Date           | Date that the maintainance event was created                                          |
+| notes     | Text           | Extra details that the user want to provide about the maintainance event                  |
+| vehicleID | Text           | Identifies the vehicle assocated with the maintainance event                              |
```
```suggestion
| vehicleID | Text           | Identifies the vehicle associated with the maintenance event                              |
```

typo

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/FirestoreDetails.md:
```diff
@@ -0,0 +1,78 @@
+| message    | Text           | Text content of the alert                                                          |
+| title      | Text           | Name given to the alert                                                            |
+
+### maintainance_events
```
```suggestion
### maintenance_events
```

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/FirestoreDetails.md:
```diff
@@ -0,0 +1,78 @@
+
+### maintainance_events
+
+Maintainance events collection contains the individual maintainance events that are associated
```
```suggestion
Maintenance events collection contains the individual maintenance events that are associated
```

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/FirestoreDetails.md:
```diff
@@ -0,0 +1,78 @@
+| id     | DocumentID     | Unique vehicle identifier that will auto-populate your objects id property |
+| userID | Text           | Identifies the user associated with the vehicle                                |
+| name | Text | Customized name given to vehicle entry
+| make | Text | Indicates the brand of the vehicle
```
```suggestion
| make | Text | Brand of the vehicle
```

mikaelacaron reviewed (approved):
The alignment was aligned for some of the tables but not others, so I fixed, that also several words are spelled wrong

I'm going to approve this, because I want to get it merged, but you didn't answer whether or not these rules work for only reading your own events / vehicles, and not being able to read other users information

mikaelacaron committed:
Fix spelling and alignment

And fix the link to the FirestoreDetails

rramirez-dev commented:
According to firebase docs https://firebase.google.com/docs/rules/rules-and-auth this gives users access to their own data.
